providers/multi/provider.go

```diff
@@ -60,21 +68,96 @@ func New(opts Options) *Provider {
 	if p.opts.Separator == "" {
 		p.opts.Separator = "#"
 	}
+	if p.opts.ChannelSize <= 0 {
+		p.opts.ChannelSize = 10
+	}
 
 	p.log = log.Log.WithName("multi-provider")
 
+	p.indexers = make([]index, 0)
 	p.providers = make(map[string]multicluster.Provider)
-	p.providerCancel = make(map[string]context.CancelFunc)
+	p.providersCancel = make(map[string]context.CancelFunc)
 
 	return p
 }
 
-// SetManager sets the manager for the provider.
-func (p *Provider) SetManager(mgr mctrl.Manager) {
-	if p.mgr != nil {
-		p.log.Error(nil, "manager already set, overwriting")
+// Start runs the provider. It runs all providers that implement
+// multicluster.ProviderRunnable, even those added after Start() has
+// been called.
+func (p *Provider) Start(ctx context.Context, aware multicluster.Aware) error {
```

Comment:
We had a quick for runnable that these methods can be ran only once. And it will be providers responsibility to ensure this. 
So is it worth adding `sync.Once` somewhere here? Maybe per multi-provider?

Reply:
I added a once but it has the usual problem that only the first invocation will get the error. 
I had a look at sync.OnceValue, but that would look something like this:

```go
type Provider struct {
	// ...
	once sync.OnceValue[error]
}
// ...
func (p *Provider) Start(ctx context.Context, aware multicluster.Aware) error {
	p.lock.Lock()
	if p.once == nil {
		p.once = sync.OnceValue(func() error {
			return p.start(ctx, aware)
		})
	}
	p.lock.Unlock()
	return p.once()
}
```

I also don't like that the context and aware from subsequent calls are ignored. If a provider is accidentally reused that could lead to opaque problems.

I think I'd prefer a `multicluster.ErrProviderAlreadyRunning` and providers having to return this error if they are already running.